test/Generator/ParameterGeneratorTest.php

```diff
@@ -236,6 +240,255 @@ public function testGeneratedParametersHaveEscapedDefaultValues()
+            $parameterName
+        ));
+
+        self::assertSame($expectedType, $parameter->getType());
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function internalReflectionHintsProvider()
+    {
+        $parameters = [
+            [InternalHintsClass::class, 'arrayParameter', 'foo', 'array'],
+            [InternalHintsClass::class, 'callableParameter', 'foo', 'callable'],
+            [InternalHintsClass::class, 'intParameter', 'foo', 'int'],
+            [InternalHintsClass::class, 'floatParameter', 'foo', 'float'],
+            [InternalHintsClass::class, 'stringParameter', 'foo', 'string'],
+            [InternalHintsClass::class, 'boolParameter', 'foo', 'bool'],
+            [ClassTypeHintedClass::class, 'selfParameter', 'foo', ClassTypeHintedClass::class],
+            [ClassTypeHintedClass::class, 'classParameter', 'foo', ClassTypeHintedClass::class],
+            [ClassTypeHintedClass::class, 'otherClassParameter', 'foo', InternalHintsClass::class],
+            [ClassTypeHintedClass::class, 'closureParameter', 'foo', \Closure::class],
+            [ClassTypeHintedClass::class, 'importedClosureParameter', 'foo', \Closure::class],
+            [DocBlockOnlyHintsClass::class, 'arrayParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'callableParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'intParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'floatParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'stringParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'boolParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'selfParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'classParameter', 'foo', null],
+            [DocBlockOnlyHintsClass::class, 'otherClassParameter', 'foo', null],
+        ];
+
+        // just re-organizing the keys so that the phpunit data set makes sense in errors:
+        return array_combine(
+            array_map(
+                function (array $definition) {
+                    return $definition[0] . '#' . $definition[1];
+                },
+                $parameters
+            ),
+            $parameters
+        );
+    }
+
+    /**
+     * @group zendframework/zend-code#29
+     *
+     * @requires PHP 7.0
+     *
+     * @dataProvider variadicHintsProvider
+     *
+     * @param string $className
+     * @param string $methodName
+     * @param string $parameterName
+     * @param string $expectedGeneratedSignature
+     */
+    public function testVariadicArgumentFromReflection(
+        string $className,
+        string $methodName,
+        string $parameterName,
+        string $expectedGeneratedSignature
+    ) {
+        $parameter = ParameterGenerator::fromReflection(new ParameterReflection(
+            [$className, $methodName],
+            $parameterName
+        ));
+
+        self::assertTrue($parameter->getVariadic());
```

Comment:
The method is static: should be used statically. I didn't change the usage elsewhere (because unrelated), but the correct approach should be taken here IMO